exporters/metric/cortex/cortex.go

```diff
@@ -264,21 +259,95 @@ func convertFromMinMaxSumCount(record metric.Record, minMaxSumCount aggregation.
 
 	// Create labels, including metric name
 	name = sanitize(record.Descriptor().Name() + "_count")
-	labels = createLabelSet(record, "__name__", name)
+	labels := createLabelSet(record, "__name__", name)
 
 	// Create TimeSeries
 	countTimeSeries := &prompb.TimeSeries{
 		Samples: []prompb.Sample{countSample},
 		Labels:  labels,
 	}
 
+	// Return all timeSeries
 	tSeries := []*prompb.TimeSeries{
 		minTimeSeries, maxTimeSeries, countTimeSeries,
 	}
 
 	return tSeries, nil
 }
 
+// convertFromDistribution returns n TimeSeries for n quantiles in a distributions
+func convertFromDistribution(record metric.Record, distribution aggregation.Distribution, quantiles []float64) ([]*prompb.TimeSeries, error) {
+	var timeSeries []*prompb.TimeSeries
+	metricName := sanitize(record.Descriptor().Name())
+
+	// For each configured quantile, get the value and create a timeseries
+	for _, q := range quantiles {
+		value, err := distribution.Quantile(q)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add quantile as a label. e.g. {quantile="0.5"}
+		quantileStr := strconv.FormatFloat(q, 'f', -1, 64)
+
+		// Create TimeSeries
+		tSeries := createTimeSeries(record, value, "__name__", metricName, "quantile", quantileStr)
+		timeSeries = append(timeSeries, tSeries)
+	}
+
+	return timeSeries, nil
+}
+
+// convertFromHistogram returns
+func convertFromHistogram(record metric.Record, histogram aggregation.Histogram) ([]*prompb.TimeSeries, error) {
+	var timeSeries []*prompb.TimeSeries
+	metricName := sanitize(record.Descriptor().Name())
+
+	// Create Sum TimeSeries
+	sum, err := histogram.Sum()
+	if err != nil {
+		return nil, err
+	}
+	sumTimeSeries := createTimeSeries(record, sum, "__name__", metricName+"_sum")
+	timeSeries = append(timeSeries, sumTimeSeries)
+
+	// Handle Histogram buckets
+	buckets, err := histogram.Histogram()
+	if err != nil {
+		return nil, err
+	}
+
+	var totalCount float64
+	// counts maps from the bucket upper-bound to the cumulative count.
+	// The bucket with upper-bound +inf is not included.
+	counts := make(map[float64]float64, len(buckets.Boundaries))
+	for i, boundary := range buckets.Boundaries {
+		// Add bucket count to totalCount and record in map
+		totalCount += buckets.Counts[i]
+		counts[boundary] = totalCount
```

Comment:
Is this map used?

Reply:
It was, but I refactored it. I'm removing it.